### PR TITLE
Task 12 

### PR DESCRIPTION
**Pull request by svetamint**
```
add validation

Acceptance criteria:

Connect the Joi library
Learn what is Middleware
Write a generic function to validate incoming parameters on requests
Add validation to existing queries

Task-13: Add nun jack for post request
Acceptance criteria:

GET request should return HTML code with form
POST request should return HTML code with photo
Nunjacks is used

added ESLint, husky with pre-commit hooks,  integrating pipeline to run linting checks on pull requests and readme.md
```

### CODE REVIEW

File: task9/validation/validator.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const validate = (schema) => {
+    return (request, response, next) => {
+        const dataToValidate = Object.keys(request.body).length > 0 ? request.body : request.query;
+        const { error, value } = schema.validate(dataToValidate, { abortEarly: false });
+        if (error) {
+            const errorMessages = error.details.map(err => err.message);
+            return response.status(400).json({ errors: errorMessages });
```
Comment (Mikhail1992) [inline task9/validation/validator.js:7]: you can throw custom Exception here

Comment (svetamint) [reply task9/validation/validator.js]: done

File: task9/exception/ValidationException.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import Exception from "./Exception.js";
+
+class ValidationException extends Exception {
```
Comment (Mikhail1992) [inline task9/exception/ValidationException.js:3]: do we really need new ValidationException? Exception should be enough

Comment (Mikhail1992) [reply task9/exception/ValidationException.js:3]: But you have 1 exception filter (error handler) which will expect limited amount of error types

Comment (svetamint) [reply task9/exception/ValidationException.js:3]: I've check exception filter and it is working for ValidationException pretty well(I debugged it), cause I also was curious will it we working or not
image

btw, I can simplify ValidationException to this implementation:

```
import Exception from './Exception.js';

class ValidationException extends Exception {
}

export default ValidationException;
```

maybe I miss smth, but errorHandler is catching Error(Exception extends Error, ValidationException extends Exception, according to transitivity property ValidationException extends Error)

Comment (svetamint) [reply task9/exception/ValidationException.js]: fixed

**Review by Mikhail1992 — APPROVED** [review]

**Pull request merged**